File: code/python/subsetsWithDup.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        res = []
        newres = []
        n = len(nums)
        nums.sort()
        for i in range(pow(2,n)):
            s = self.binToString(bin(i),n)
            temp = []
            for j in range(n):
                if s[n-1-j] == "1":
                    temp.append(nums[j])
            if temp not in res:
                res.append(temp)

        return res
    
    def binToString(self,num:bin,n:int) -> str:
        s = (str(num))[2:]
        length = len(s)
        for i in range(n-length):
            s = "0" + s;
        return s        

File: code/java/subsetsWithDup.java
class Solution {
    public List<List<Integer>> subsetsWithDup(int[] nums) {
        List<List<Integer>> res = new ArrayList<List<Integer>>();
        int n = nums.length;
        Arrays.sort(nums);
        for(int x = 0;x<Math.pow(2,n);x++){
            List<Integer> temp = new ArrayList<Integer>();
            String s = binToString(x,n);
            for(int j=0;j<s.length();j++)
                if(s.charAt(n-1-j) == '1')
                    temp.add(nums[j]);
            if(!res.contains(temp))
                res.add(temp);
        }
        return res;
           
    }
    
    public String binToString(int x,int n){
        String s = Integer.toBinaryString(x);
        int len = s.length();
        for(int i=0;i<n-len;i++)
            s = "0" + s;
        return s;
    }
}
